metacatalog/test/test_api_add_find.py

```diff
@@ -271,13 +271,26 @@ def add_details(session):
     e2.details_table(fmt='markdown')
 
     # find the details
+    # find by key and value
     found_entry = api.find_entry(session, details=dict(answer=42))[0]
     assert e2.id == found_entry.id
 
-    # find nested details
-    found_entry2 = api.find_entry(session, details=dict(foo=dict(baz=42)))[0]
+    # find by key (wildcard) and value
+    found_entry2 = api.find_entry(session, details={'a*wer':42})[0]
     assert e2.id == found_entry2.id
 
+    # find by key and value (wildcard)
+    found_entry3 = api.find_entry(session, details={'banana':'both *e it'})
+    assert len(found_entry3) == 2
+
+    # find nested details
+    found_entry4 = api.find_entry(session, details=dict(foo=dict(baz=42)))[0]
+    assert e2.id == found_entry4.id
+
+    # find by value only
+    found_entry4 = api.find_entry(session, details={'*': 'both *ove it'})

```

Comment:
how will this be translated into SQL? does it actually add a wildcard search on ALL keys, or is the query built to just match the value?
Could again have huge performance implications.

Reply:
API search:
```python
api.find_entry(session, details={'*': 'both *ove it'}, return_iterator=True)
```

SQL:
```sql
SELECT entries.id, entries.uuid, entries.title, entries.abstract, entries.external_id, ST_AsEWKB(entries.location) AS location, ST_AsEWKB(entries.geom) AS geom, entries.version, entries.latest_version_id, entries.is_partial, entries.comment, entries.citation, entries.license_id, entries.variable_id, entries.datasource_id, entries.embargo, entries.embargo_end, entries.publication, entries."lastUpdate" 
FROM entries JOIN details ON entries.id = details.entry_id 
WHERE entries.is_partial = false AND entries.latest_version_id IS NULL AND details.stem LIKE '%%' AND details.raw_value ->> '__literal__' LIKE 'both %%ove it'
```

It looks like it is actually a wildcard search on ALL keys, but only where `raw_value['__literal__']` matches the searched value:
`((stem)::text ~~ '%%'::text) AND ((raw_value ->> '__literal__'::text) ~~ 'both %%ove it'::text))`